### PR TITLE
Rename _start to __CPROVER_start

### PR DESCRIPTION
**Pull request by tautschnig**
While identifiers starting with _ (or __ ?) are reserved according to the C standard,
people nevertheless use them. Using __CPROVER_start will make collisions less
likely.

Includes cleanup so that, outside cegis, there is exactly one point where this
name is spelled out.

### CODE REVIEW

File: src/cegis/invariant/fitness/concrete_fitness_source_provider.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ bool contains(const std::string &haystack, const std::string &needle)
 
 bool handle_start(std::string &source, const std::string &line)
 {
-  if ("void _start(void)" != line) return false;
+  if ("void __CPROVER_start(void)" != line) return false;
```
Comment (peterschrammel) [inline src/cegis/invariant/fitness/concrete_fitness_source_provider.cpp]: http://cbmc-linter.diffblue.com.s3-website-us-east-1.amazonaws.com/linter-9845351f4f89cef764906458b2d5ff8e8b0e9919.txt

File: src/cegis/jsa/genetic/jsa_source_provider.cpp [hunk 2]
```diff
@@ -26,7 +26,7 @@ jsa_source_providert::jsa_source_providert(jsa_symex_learnt &lcfg) :
 {
 }
 
-#define START_METHOD_PREFIX "void _start"
+#define START_METHOD_PREFIX "void __CPROVER_start"
```
Comment (peterschrammel) [inline src/cegis/jsa/genetic/jsa_source_provider.cpp]: Does it make sense to put this into util/prefix.h to really have it in a single place?

**Comment by martin-cs** [comment]
> -#define START_METHOD_PREFIX "void _start"
> +#define START_METHOD_PREFIX "void __CPROVER_start"
> 
> Does it make sense to put this into util/prefix.h to really have it in a single place?

Perhaps as part of a general clean up of working out which variables,
functions, locations, etc. are internal?

**Comment by tautschnig** [comment]
There now really is just one point where the name is generated, which is also used from cegis.

**Comment by peterschrammel** [comment]
Just observing that we have 5 instances of "__CPROVER_initialize" and 3 instances of CPROVER_PREFIX "initialize" in the code. I'd prefer the latter...

**Comment by tautschnig** [comment]
I've created #735 to track cleanup of other uses of `"__CPROVER_...` - 181 cases, which I don't have time for on the spot.

**Pull request merged**